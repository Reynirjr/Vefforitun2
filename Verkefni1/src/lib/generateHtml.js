
  import { stoppaHTML } from './stoppaHtml.js';
  
  /**
 * Skrifar HTML fyrir flokk
 * @param {object} data staðfest gögn
 * @returns {string} HTML strengur sem við skrifum 
 */
 export function geraFlokkHtml(data) {
    const questionsHtml = data.questions
      .map((q) => {
        const answersHtml = q.answers
          .map((a) => `
          <li> 
            <button 
              class= "answer"
              data-correct="${a.correct}"
            >
            ${stoppaHTML(a.answer)}
            </button>
          </li>`
          )
          .join('');
  
        return /* html */ `
          <div class = "spurning">
            <h2>${stoppaHTML(q.question)}</h2>
            <ul>
              ${answersHtml}
            </ul>
          </div>
        `;
      })
      .join('');
  
    return /* html */ `
  <!DOCTYPE html>
  <html lang="is">
  <head>
    <meta charset="utf-8">
    <title>${data.title}</title>
    <link rel="stylesheet" href="./styles.css">
  </head>
  <body>
    <h1>${data.title}</h1>
    ${questionsHtml}
    <footer>
      <a href="./index.html">Aftur á forsíðu</a>
    </footer>
      <script>
        document.addEventListener('DOMContentLoaded', () => {
          const answerButtons = document.querySelectorAll('.answer');

          answerButtons.forEach((btn) => {
            btn.addEventListener('click', (e) => {
              const isCorrect = e.currentTarget.dataset.correct === 'true';
              if (isCorrect) {
                e.currentTarget.classList.add('correct');
              } else {
                e.currentTarget.classList.add('wrong');
              }
            });
          });
        });
      </script>
  </body>
  </html>
    `;
  }