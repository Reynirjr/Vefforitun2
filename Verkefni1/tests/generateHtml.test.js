import { describe, it, expect } from 'vitest';
import { geraFlokkHtml } from '../src/lib/generateHtml.js';

describe('geraFlokkHtml', () => {
  it('generates HTML with questions and answers', () => {
    const data = {
      title: 'Title',
      questions: [
        {
          question: 'Q1?',
          answers: [
            { answer: 'A1', correct: true },
            { answer: 'A2', correct: false },
          ],
        },
      ],
    };

    const html = geraFlokkHtml(data);

    expect(html).toContain('<h1>Title</h1>');
    expect(html).toContain('Q1?');
    expect(html).toContain('A1');
    expect(html).toContain('A2');
    expect(html).toContain('data-correct="true"');
    expect(html).toContain('data-correct="false"');
  });

  it('matches snapshot', () => {
    const data = {
      title: 'Snapshot Title',
      questions: [
        {
          question: 'Snapshot Q?',
          answers: [{ answer: 'Snapshot A1', correct: true }],
        },
      ],
    };
    const html = geraFlokkHtml(data);
    expect(html).toMatchSnapshot();
  });
});
