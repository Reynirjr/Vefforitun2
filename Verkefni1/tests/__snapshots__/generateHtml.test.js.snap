// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`geraFlokkHtml > matches snapshot 1`] = `
"
  <!DOCTYPE html>
  <html lang="is">
  <head>
    <meta charset="utf-8">
    <title>Snapshot Title</title>
    <link rel="stylesheet" href="./styles.css">
  </head>
  <body>
    <h1>Snapshot Title</h1>
    
          <div class = "spurning">
            <h2>Snapshot Q?</h2>
            <ul>
              
          <li> 
            <button 
              class= "answer"
              data-correct="true"
            >
            Snapshot A1
            </button>
          </li>
            </ul>
          </div>
        
    <footer>
      <a href="./index.html">Aftur á forsíðu</a>
    </footer>
      <script>
        document.addEventListener('DOMContentLoaded', () => {
          const answerButtons = document.querySelectorAll('.answer');

          answerButtons.forEach((btn) => {
            btn.addEventListener('click', (e) => {
              const isCorrect = e.currentTarget.dataset.correct === 'true';
              if (isCorrect) {
                e.currentTarget.classList.add('correct');
              } else {
                e.currentTarget.classList.add('wrong');
              }
            });
          });
        });
      </script>
  </body>
  </html>
    "
`;
